[Tan Jean Yang] iP

jeanyang04 opened the pull request:
# Skibidi

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](#)

Skibidi frees your mind of having to remember things you need to do. It's:

- text-based
- easy to learn
- **FAST** _SUPER FAST_ to use

All you need to do is,

1. download it from [here](#).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😊

And it is **FREE**!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jeanyang04 committed:
Change file name and add greeting message

jeanyang04 committed:
Add deliverables for Level-1

jeanyang04 committed:
Add app storage

jeanyang04 committed:
Add 'add' and 'list' command

jeanyang04 committed:
Merge tag 'Level-2'

jeanyang04 committed:
Add task class

jeanyang04 committed:
Add support for task type

jeanyang04 committed:
Add support for mark and unmark

jeanyang04 committed:
Fix invalid input crash bug

jeanyang04 committed:
Add todo class

jeanyang04 committed:
Add deadline class

jeanyang04 committed:
Add event class

jeanyang04 committed:
Add todo and deadline

jeanyang04 committed:
Add event class and clean up parser code

jeanyang04 committed:
Remove commented code

jeanyang04 committed:
Add text UI testing

jeanyang04 committed:
Add error handling

jeanyang04 committed:
Add deleteCommand class

jeanyang04 committed:
Add support for delete command

jeanyang04 committed:
Add storageManager class

jeanyang04 committed:
Change task classes to support storage feature

jeanyang04 committed:
Delete addCommand class

jeanyang04 committed:
Change constructor calls in command classes

jeanyang04 committed:
bugFix: Handle IOException in Skibidi.java

jeanyang04 committed:
bugFix: Cannot load isDone upon start

jeanyang04 committed:
Add date support for deadline

jeanyang04 committed:
Add support for date & time for event

jeanyang04 committed:
Merge branch 'branch-Level-8'

jeanyang04 committed:
Make line divider into a String variable

jeanyang04 committed:
Merge branch 'branch-A-MoreOOP'

jeanyang04 committed:
Refactor common package and move Skibidi into package

jeanyang04 committed:
Merge branch 'branch-A-Packages'

jeanyang04 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jeanyang04 committed:
Set up gradle build and run

jeanyang04 committed:
Merge branch 'branch-A-Gradle'

jeanyang04 committed:
Add two tesst class

jeanyang04 committed:
Merge branch 'branch-A-JUnit'

jeanyang04 committed:
Create jar file

jeanyang04 committed:
Merge branch 'branch-A-Jar'

jeanyang04 committed:
Add JavaDoc comments

jeanyang04 committed:
Change some fields to final

jeanyang04 committed:
Add find command file

jeanyang04 committed:
Add find task by keyword method

jeanyang04 committed:
Change FindCommand execute method

jeanyang04 committed:
Print find result in list

jeanyang04 committed:
Merge branch 'branch-A-JavaDoc'

jeanyang04 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/skibidi/Ui.java

jeanyang04 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/storage/Task.java

jasmiinee commented on src/main/java/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,59 @@
+import storage.Deadline;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
```
Should these java imports be moved to the top instead?

jasmiinee commented on src/main/java/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,83 @@
+import storage.Event;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
```
Should these java imports also be at the top instead?

jasmiinee commented on src/main/java/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,51 @@
+            Task task = storage.getTask(index);
+            storage.deleteTask(index);
+            ui.printMessage("Noted. I've removed this task:\n  " + task +
+                    "\nNow you have " + storage.getTaskCount() + " tasks in the list.");
```
Should the line break be before the '+' operator instead?

jasmiinee commented on src/main/java/skibidi/Parser.java:
```diff
@@ -0,0 +1,42 @@
+package skibidi;
+
+import commands.*;
```
Should the imports be listed explicitly instead of using the '*' operator?

jasmiinee commented on src/main/java/storage/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+package storage;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
```
Should these imports be in lexicographic order instead?

jasmiinee commented on src/main/java/storage/Event.java:
```diff
@@ -0,0 +1,46 @@
+package storage;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
```
Should these imports be in lexicographic order instead?

jasmiinee commented on src/main/java/storage/Event.java:
```diff
@@ -0,0 +1,46 @@
+     */
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + from.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm")) + " to: " + to.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm")) + ")";
```
Should this line be wrapped instead?

jasmiinee commented on src/main/java/storage/StorageManager.java:
```diff
@@ -0,0 +1,108 @@
+package storage;
+
+import java.io.*;
```
Should the imports be listed out explicitly instead?

jasmiinee commented on src/main/java/storage/TaskStorage.java:
```diff
@@ -0,0 +1,112 @@
+        }
+        return tasks.get(index);
+    }
+
```
Possible JavaDoc comment here?

jasmiinee commented on src/main/java/storage/Task.java:
```diff
@@ -0,0 +1,64 @@
+     * @return The task in the file format.
+     */
+    public abstract String toFileFormat();
+}
```
i like how clean this code is!

jasmiinee commented on src/test/java/commands/AddDeadlineCommandTest.java:
```diff
@@ -0,0 +1,52 @@
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Should these static imports be at the top of the package groupings instead?

jasmiinee commented on src/test/java/commands/AddTodoCommandTest.java:
```diff
@@ -0,0 +1,44 @@
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Should these static imports be at the top of the package groupings instead?

jasmiinee commented on src/main/java/commands/AddTodoCommand.java:
```diff
@@ -0,0 +1,51 @@
+import storage.TaskStorage;
+import storage.Todo;
+
+import java.io.IOException;
```
Should this java import be at the top instead? I noticed the same issue in several other places too.

jasmiinee reviewed (commented):
Overall, code looks pretty good and neat except for a few coding standard violations :)

jiaxinnns commented on src/main/java/skibidi/Parser.java:
```diff
@@ -0,0 +1,42 @@
+
+        String normalizedInput = input.trim().toLowerCase();
+
+        if (normalizedInput.equals("bye")) {
```
Can a switch statement or Enum be used to handle these if-else statements to make your code cleaner?

jiaxinnns commented on src/main/java/commands/FindCommand.java:
```diff
@@ -0,0 +1,31 @@
+        if (matchingTasks.isEmpty()) {
+            ui.printMessage("No matching tasks found.");
+        } else {
+            StringBuilder str = new StringBuilder("Here are the matching tasks in your list:\n");
```
Could the variable name be a little more descriptive?

jiaxinnns commented on src/main/java/commands/ListCommand.java:
```diff
@@ -0,0 +1,33 @@
+     */
+    @Override
+    public boolean execute(Ui ui, TaskStorage storage) {
+        StringBuilder list = new StringBuilder();
```
Could the variable list be a little more descriptive as well? It might help with the readability of your code

jiaxinnns commented on src/main/java/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,59 @@
+    public AddDeadlineCommand(String input) throws SkibidiException{
+        String[] parts = input.split("/by ");
+        if (parts.length < 2) {
+            throw new SkibidiException("Invalid deadline format. Usage: deadline [description] /by [date]");
```
I like how you've given the user the expected input format in the error message!

jiaxinnns commented on src/main/java/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,83 @@
+        this.description = partsFrom[0].substring(6).trim();
+        this.startTime = parseDateTime(partsTo[0].trim());
+        this.endTime = parseDateTime(partsTo[1].trim());
+
```
There are a few blocks of empty lines here and there, just a small issue but worth taking note!

jiaxinnns commented on src/main/java/storage/StorageManager.java:
```diff
@@ -0,0 +1,108 @@
+        boolean isDone = parts[1].trim().equals("1");
+        String description = parts[2];
+
+        switch (type) {
+            case "T":
+                return new Todo(description, isDone);
+            case "D":
+                String by = parts[3];
+                return new Deadline(description, by, isDone);
+            case "E":
+                String from = parts[3];
+                String to = parts[4];
+                return new Event(description, from, to, isDone);
+            default:
+                throw new IllegalArgumentException("Invalid task type in file.");
+        }
```
Maybe the indentation can be changed since cases within a switch block should not be indented

jiaxinnns reviewed (commented):
Minor code convention violations here and there but I really like how you've organised your classes! It's really neat and readable

jeanyang04 committed:
Fix style using checkstyle

jeanyang04 committed:
Merge branch 'branch-A-CheckStyle'

jeanyang04 committed:
Add javafx

jeanyang04 committed:
Change UI methods to return String

jeanyang04 committed:
Change execute method for command

jeanyang04 committed:
Remove extra line of response from Ui.java

jeanyang04 committed:
Merge branch 'branch-Level-10'

jeanyang04 committed:
Add assertions to Skibidi and DeadlineCmd

- Added assertions in `Skibidi` constructor to ensure `Ui`, `Parser`, and `TaskStorage` are not null.
- Added assertions in `AddDeadlineCommand` constructor to validate non-null and non-empty description and deadline.
- Added assertions in `AddDeadlineCommand.execute` method to ensure `Ui` and `TaskStorage` are not null.

jeanyang04 committed:
Add assertions to all Add* commands

- Added assertions in `AddEventCommand` constructor to validate non-null and non-empty description, start time, and end time.
- Added assertions in `AddEventCommand.execute` method to ensure `Ui` and `TaskStorage` are not null.
- Added assertions in `AddTodoCommand` constructor to validate non-null and non-empty description.
- Added assertions in `AddTodoCommand.execute` method to ensure `Ui` and `TaskStorage` are not null.

jeanyang04 committed:
Add assertions to Delete,Echo Command classes

- Added assertions in `DeleteCommand.execute` method to ensure `Ui` and `TaskStorage` are not null.
- Added assertions in `EchoCommand.execute` method to ensure `Ui` is not null.

jeanyang04 committed:
Add assertions to Find, List, Exit Command classes

- Added assertions in `FindCommand.execute` method to ensure `Ui` and `TaskStorage` are not null.
- Added assertions in `ListCommand.execute` method to ensure `Ui` and `TaskStorage` are not null.
- Added assertions in `ExitCommand.execute` method to ensure `Ui` and `TaskStorage` are not null.

jeanyang04 committed:
Refactor a few Command files

jeanyang04 committed:
Refactor other classes

jeanyang04 committed:
Refactor echo and mainWindow

jeanyang04 committed:
Merge pull request #1 from jeanyang04/branch-A-Assertions

Branch a assertions

jeanyang04 committed:
Merge branch 'master' into branch-A-CodeQuality

jeanyang04 committed:
Merge pull request #2 from jeanyang04/branch-A-CodeQuality

Branch a code quality

jeanyang04 committed:
Add addTag and removeTag command

jeanyang04 committed:
Update parser to support tag

jeanyang04 committed:
Fix remove tag bug

jeanyang04 committed:
Fix bug of tags not saving to local

jeanyang04 committed:
Merge branch 'branch-C-Tagging'

jeanyang04 committed:
Add Ai.md

jeanyang04 committed:
Merge branch 'branch-A-AiAssisted'

jeanyang04 committed:
Fix empty tag and empty echo crash

jeanyang04 committed:
Fix mark and unmark crash and exit command

jeanyang04 committed:
Merge branch 'bugFix'

jeanyang04 committed:
Fix tagging issues

jeanyang04 committed:
Merge branch 'bugFix'

jeanyang04 committed:
Update user guide

jeanyang04 committed:
Merge branch 'A-UserGuide'

jeanyang04 committed:
Edit readme

jeanyang04 committed:
Update screenshot

- Had to commit on master due to unknown bug
